internal/venice-common/src/main/java/com/linkedin/venice/serialization/avro/AvroSpecificStoreDeserializerCache.java

```diff
@@ -0,0 +1,52 @@
+package com.linkedin.venice.serialization.avro;
+
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.schema.SchemaReader;
+import com.linkedin.venice.serialization.StoreDeserializerCache;
+import com.linkedin.venice.serializer.FastSerializerDeserializerFactory;
+import com.linkedin.venice.serializer.RecordDeserializer;
+import com.linkedin.venice.utils.SparseConcurrentList;
+import java.util.function.IntFunction;
+import org.apache.avro.Schema;
+import org.apache.avro.specific.SpecificRecord;
+
+
+public class AvroSpecificStoreDeserializerCache<V extends SpecificRecord> implements StoreDeserializerCache<V> {
+  private final IntFunction<Schema> schemaGetter;
+  private final SparseConcurrentList<RecordDeserializer<V>> cache;
+  private final Class<V> valueClass;
+
+  public AvroSpecificStoreDeserializerCache(SchemaReader schemaReader, Class<V> valueClass) {
+    this(schemaReader::getValueSchema, valueClass);
+  }
+
+  public AvroSpecificStoreDeserializerCache(
+      ReadOnlySchemaRepository schemaRepository,
+      String storeName,
+      Class<V> valueClass) {
+    this(id -> schemaRepository.getValueSchema(storeName, id).getSchema(), valueClass);
+  }
+
+  private AvroSpecificStoreDeserializerCache(IntFunction<Schema> schemaGetter, Class<V> valueClass) {
+    this.schemaGetter = schemaGetter;
+    this.cache = new SparseConcurrentList<>();
+    this.valueClass = valueClass;
+  }
+
+  @Override
+  public RecordDeserializer<V> getDeserializer(int writerSchemaId, int readerSchemaId) {
```

Comment:
I tried it but it will cause errors and require refactoring of ChunkingAdapter since it was originally implemented when both specific and generic deser cache was sharing the same class and invoked with the same `getDeserializer(int writerSchemaId, int readerSchemaId)`. We can do it in another PR for the refactor since this PR is getting bigger and bigger.